### PR TITLE
feat(lspsaga): support v0.2.7

### PR DESCRIPTION
**Pull request by Jint-lzxy**
This PR added support for lspsaga@v0.2.7. Changes include:
- Implemented new highlight groups;
- The `LspKind*` highlight groups are used to set the colors for lspkind icons **(finally!)**
  - Note: Those highlights do _not_ belong to any integration. I added them to support "dynamically changing flavors when editing". Not sure if this is the best way to do so.

It'll be marked as ready for review after changes are merged upstream 👍

### CODE REVIEW

**Comment by nullchilly** [comment]
In favour of directly using `SagaWinbarClass = { link = "Class }` instead of `custom_kind` because lspsaga makes no effect to preserve highlight groups on ColorScheme event

**Comment by Jint-lzxy** [comment]
> In favour of directly using `SagaWinbarClass = { link = "Class }` instead of `custom_kind` because lspsaga makes no effect to preserve highlight groups on ColorScheme event

@nullchilly Sorry, I can't quite understand that. Now `lspsaga` uses highlight groups to which the `lspkind` icons are linked to set all its colors. If we only set highlights for `winbar`, the icons for other features _(such as outline)_ will have no color then. IMO `lspsaga` can handle this correctly now, see: [lspkind.lua#L110](https://github.com/glepnir/lspsaga.nvim/blob/match_theme/lua/lspsaga/lspkind.lua#L110).

Did I miss something here?

File: lua/catppuccin/groups/integrations/lsp_saga.lua [hunk 1]
```diff
@@ -24,42 +25,89 @@ function M.get()
...
-		LspSagaWinbarStruct = { fg = C.mauve },
-		LspSagaWinbarEvent = { fg = C.mauve },
-		LspSagaWinbarOperator = { fg = C.green },
-		LspSagaWinbarTypeParameter = { fg = C.green },
-		LspSagaWinbarTypeAlias = { fg = C.green },
-		LspSagaWinbarParameter = { fg = C.blue },
-		LspSagaWinbarStaticMethod = { fg = C.peach },
-		LspSagaWinbarMacro = { fg = C.red },
+		-- Lspkind icons support
+		LspKindClass = { fg = C.yellow },
```
Comment (nullchilly) [inline lua/catppuccin/groups/integrations/lsp_saga.lua:35]: I mean we directly set it like this instead:

```lua
SagaWinbarClass = { link = "Class }
```

File: lua/catppuccin/groups/integrations/lsp_saga.lua [hunk 2]
```diff
@@ -24,42 +25,89 @@ function M.get()
...
+		-- Microsoft-specific icons.
+		LspKindText = { fg = C.green },
+		LspKindSnippet = { fg = C.mauve },
+		LspKindFolder = { fg = C.blue },
+		LspKindUnit = { fg = C.green },
+		LspKindValue = { fg = C.peach },
+	}
+end
+
+function M.custom_kind()
```
Comment (nullchilly) [inline lua/catppuccin/groups/integrations/lsp_saga.lua:74]: So there's no need for user to manually add this to their configuration

Comment (Jint-lzxy) [reply lua/catppuccin/groups/integrations/lsp_saga.lua:74]: But if we set it this way colors are limited to `winbar` _only_, and there would be no highlight in other places that also utilizes these icons like "lspsaga outline".

Comment (nullchilly) [reply lua/catppuccin/groups/integrations/lsp_saga.lua:74]: Ah yeah sorry I overlooked the outline feature

**Review by nullchilly — APPROVED** [review]

File: lua/catppuccin/groups/integrations/lsp_saga.lua (hunk 1 above)
Comment (Jint-lzxy) [reply lua/catppuccin/groups/integrations/lsp_saga.lua:35]: Maybe they can be changed to:
```lua
LspKindClass = { link = "Class" },
```
?

**Pull request merged**